src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs

```diff
@@ -15,242 +14,273 @@
+            previousContentWasStringLiteralExpression = currentContentIsStringOrCharacterLiteral;
+        }
+
+        return generator.InterpolatedStringExpression(startToken, content, endToken);
+
+        SyntaxNode TryRemoveToString(SyntaxNode piece)
+        {
+            if (supportsInterpolatedStringHandler)
+            {
+                // if it's a call to object's .ToString (or any override), then we can remove this if the runtime
+                // supports interpolated string handlers. This gies the most flexibility to the handler to decide what
+                // it wants to do.
+                if (syntaxFacts.IsInvocationExpression(piece))
                 {
-                    // "piece" is itself an interpolated string (of the same "verbatimity" as the new interpolated string)
-                    // "a" + $"{1+ 1}" -> instead of $"a{$"{1 + 1}"}" inline the interpolated part: $"a{1 + 1}"
-                    syntaxFacts.GetPartsOfInterpolationExpression(piece, out var _, out var contentParts, out var _);
-                    foreach (var contentPart in contentParts)
+                    syntaxFacts.GetPartsOfInvocationExpression(piece, out var memberAccess, out var argumentList);
+                    if (syntaxFacts.IsMemberAccessExpression(memberAccess))
                     {
-                        // Track the state of currentContentIsStringOrCharacterLiteral for the inlined parts
-                        // so any text at the end of piece can be merge with the next string literal:
-                        // $"{1 + 1}a" + "b" -> "a" and "b" get merged in the next "pieces" loop
-                        currentContentIsStringOrCharacterLiteral = syntaxFacts.IsInterpolatedStringText(contentPart);
-                        if (currentContentIsStringOrCharacterLiteral && previousContentWasStringLiteralExpression)
+                        syntaxFacts.GetPartsOfMemberAccessExpression(memberAccess, out var expression, out var name);
+                        if (syntaxFacts.GetIdentifierOfSimpleName(name).ValueText == nameof(ToString))
                         {
-                            // if piece starts with a text and the previous part was a string, merge the two parts (see also above)
-                            // "a" + $"b{1 + 1}" -> "a" and "b" get merged
-                            var newText = ConcatenateTextToTextNode(generator, content.Last(), contentPart.GetFirstToken().Text, contentPart.GetFirstToken().ValueText);
-                            content[^1] = newText;
-                        }
-                        else
-                        {
-                            content.Add(contentPart);
-                        }
+                            var symbol = semanticModel.GetSymbolInfo(memberAccess, cancellationToken).Symbol as IMethodSymbol;
+                            while (symbol?.OverriddenMethod != null)
```

Comment:
Would checking that this is reference type (object-derived) be an option instead of walking the overridden method chain?

Reply:
No. Because there might be a different ToString overload method that was being called. We need to make sure it's the actual object.ToString method (or some override of it).

Reply:
OK, just so I understand, checking against a no-arg method named ToString that is on a reference type isn't sufficient?

Reply:
Correct. You can have non-override (`new`) methods with the same name/sig.